packages/wiz-ui-react/src/components/base/navigation/components/navigation-item.tsx

```diff
@@ -1,52 +1,72 @@
 import { ComponentName } from "@wizleap-inc/wiz-ui-constants";
 import {
-  navigationItemStyle,
-  navigationItemActiveStyle,
   navigationItemIconStyle,
   navigationItemIconActiveStyle,
   navigationItemTextStyle,
   navigationItemTextActiveStyle,
-  navigationItemDisabledStyle,
   navigationItemIconDisabledStyle,
   navigationPopupContainerStyle,
+  navigationItemStyle,
+  navigationItemActiveStyle,
+  navigationItemDisabledStyle,
 } from "@wizleap-inc/wiz-ui-styles/bases/navigation.css";
 import clsx from "clsx";
-import { FC, useCallback, useRef } from "react";
+import { ComponentProps, ElementType, useCallback, useRef } from "react";
 
 import { TIcon, WizPopup, WizPopupButtonGroup, WizTooltip } from "@/components";
 
 import { ButtonGroupItem } from "../../popup-button-group/types";
 
-interface Props {
+type Props<T extends ElementType> = {
   icon: TIcon;
   label: string;
   active: boolean;
-  href: string;
   disabled?: boolean;
   tooltipText?: string;
   isPopupLocking?: boolean;
   buttons?: ButtonGroupItem[];
   isPopupOpen?: boolean;
   onTogglePopup: (isPopup: boolean) => void;
   onTogglePopupLocking: (lock: boolean) => void;
-}
-
-const NavigationItem: FC<Props> = ({
+} & (
+  | { href: string }
+  | {
+      as: T;
+      asProps: ComponentProps<T>;
```

Comment:
申し訳ありません、例示した型が不十分で `href` と `as` が同時に指定できていました。排他的な型にするためには以下のようにする必要がありそうです。

```suggestion
  | {
      href: string;
      as?: never;
      asProps?: never;
    }
  | {
      href?: never;
      as: T;
      asProps: ComponentProps<T>;
```